src/main/java/com/redhat/cloud/policies/engine/db/repositories/PoliciesRepository.java

```diff
@@ -44,22 +48,36 @@ public List<Policy> getEnabledPolicies(String accountId) {
     @CacheResult(cacheName = ACCOUNT_LATEST_UPDATE_CACHE_NAME)
     LocalDateTime findLatestUpdate(String accountId) {
         LOGGER.debugf("Finding latest policies update time for account %s", accountId);
-        String hql = "SELECT latest FROM AccountLatestUpdate WHERE accountId = :accountId";
         try {
-            return statelessSessionFactory.getCurrentSession().createQuery(hql, LocalDateTime.class)
-                    .setParameter("accountId", accountId)
-                    .getSingleResult();
+            if (orgIdConfig.isUseOrgId()) {
+                String hql = "SELECT latest FROM OrgIdLatestUpdate WHERE orgId = :orgId";
+                return statelessSessionFactory.getCurrentSession().createQuery(hql, LocalDateTime.class)
+                        .setParameter("orgId", accountId)
```

Comment:
The `orgId` should be used here, not the `accountId`. As I mentioned earlier, it would be easier to duplicate the `getEnabledPolicies` method and pass the `orgId` to the new method. The `orgIdConfig.isUseOrgId()` condition would determine which method should be called.